src/Swift.Bindings/src/Program.cs

```diff
@@ -92,18 +88,22 @@ public static void GenerateBindings(string swiftAbiPath, string dylibPath, strin
             // Initialize the Swift ABI parser
             var swiftParser = new SwiftABIParser(swiftAbiPath, dylibPath, typeDatabase, verbose);
             var moduleName = swiftParser.GetModuleName();
-            
+
             // Skip if the module has already been processed
+            // Modules will have to be processed in topological order
             if (!typeDatabase.IsModuleProcessed(moduleName))
             {
-            
+
                 // Register the module and set the filter
                 var moduleRecord = typeDatabase.Registrar.RegisterModule(moduleName);
                 moduleRecord.Path = dylibPath;
 
                 // Parse the Swift ABI file and generate declarations
                 var decl = swiftParser.GetModuleDecl();
 
+                var moduleProcessor = new ModuleProcessor(moduleName, dylibPath, decl.Types, typeDatabase);
+                moduleProcessor.FinalizeTypeProcessing();
```

Comment:
If `ModuleProcessor` is exclusively creating a type registrar, should it be returned here?

Reply:
This will be addressed in a follow-up. I need to remove logic using it from ABI parsing